sdk/scheduler/src/main/java/com/mesosphere/sdk/dcos/secrets/AlreadyExistsException.java

```diff
@@ -0,0 +1,7 @@
+package com.mesosphere.sdk.dcos.secrets;
+
+/**
+ * An operation is attempting to create a secret that already exists.
+ */
+public class AlreadyExistsException extends SecretsException {
```

Comment:
IMO just have a single `SecretsException` class with a `reason` field. Makes it easier to log the cause of the exception without involving reflection. See `PersisterException` for an example of using a `reason` field. Also, the `store` and `path` fields being passed in the 401 case could just always be passed to `SecretsException` as they give more context as to the originating request.